parsons/google/google_bigquery.py

```diff
@@ -1388,6 +1388,93 @@ def extract(
 
         logger.info(f"Finished exporting query result to {gs_destination}.")
 
+    def copy_between_projects(self, source_project, source_dataset, source_table,
+                              destination_project, destination_dataset, destination_table,
+                              if_dataset_not_exists='fail', if_table_exists='fail'):
+        """
+        Copy a table from one project to another. Set the flags if_dataset_not_exists and
+        if_table_exists to control behavior if target dataset for the table does not exist
+        and if the target table exists (default is to fail in both cases).
+        Fails if the source or target project does not exist.
+
+        `Args`:
+            source_project: str
+                Name of source project
+            source_dataset: str
+                Name of source dataset
+            source_table: str
+                Name of source table
+            destination_project: str
+                Name of destination project
+            destination_dataset: str
+                Name of destination dataset
+            destination_table: str
+                Name of destination table
+            if_dataset_not_exists: str
+                Action if dataset doesn't exist {'fail','create'}
+            if_table_exists: str
+                Action if table exists {'fail', 'overwrite'}
+
+        `Returns:`
+            None
+        """
+
+        from google.cloud import bigquery
+        from google.cloud.exceptions import NotFound
+        import sys
+
+        destination_table_id = (destination_project + "."
+                                + destination_dataset + "."
+                                + destination_table)
+        source_table_id = (source_project + "."
+                           + source_dataset + "."
+                           + source_table)
+
+        # check if destination dataset exists
+        try:
+            dataset_id = destination_project + "." + destination_dataset
+            self.client.get_dataset(dataset_id)  # Make an API request.
+            # if it exists: continue
+        except NotFound:
+            # if it doesn't exist: check the flag for creating it
+            if if_dataset_not_exists == 'create':  # create a new dataset in the destination
+                dataset = bigquery.Dataset(dataset_id)
+                dataset = self.client.create_dataset(dataset, timeout=30)
+            else:  # fail
+                print("BigQuery copy failed")
```

Comment:
I think it makes sense to use the logger here (and throughout the method) rather than print. [Here's](https://github.com/move-coop/parsons/blob/46e551edae0739007c682c70559f8a23532bbfee/parsons/google/google_bigquery.py#L110) an example of the logger in this file - if you've not used it before and are unfamiliar with log levels happy to explain a bit more how they work.

Reply:
Thanks--this looks like an error level (from what I know)--does that sound right?